### PR TITLE
feat(api): expose public REST API 

### PR DESCRIPTION
**Pull request by 0xGabi**

### CODE REVIEW

File: packages/api/src/routers/stats/getBlockDailyStats.ts [hunk 1]
```diff
@@ -8,8 +8,8 @@ export const getBlockDailyStats = timeFrameProcedure
     openapi: {
       method: "GET",
       path: `/${BLOCK_BASE_PATH}`,
-      tags: ["blocks"],
-      summary: "Get blocks time series stats",
+      tags: ["stats"],
+      summary: "get blocks time series stats",
```
Comment (PJColombo) [inline packages/api/src/routers/stats/getBlockDailyStats.ts:12]:
```suggestion
      summary: "retrieves blocks time series stats.",
```

File: packages/api/src/routers/stats/getBlockOverallStats.ts [hunk 2]
```diff
@@ -42,8 +42,8 @@ export const getBlockOverallStats = publicProcedure
     openapi: {
       method: "GET",
       path: `/${BLOCK_BASE_PATH}/overall`,
-      tags: ["blocks"],
-      summary: "Get blocks overall stats",
+      tags: ["stats"],
+      summary: "get blocks overall stats",
```
Comment (PJColombo) [inline packages/api/src/routers/stats/getBlockOverallStats.ts:46]:
```suggestion
      summary: "retrieves blocks overall stats.",
```

File: packages/api/src/routers/stats/getTransactionDailyStats.ts [hunk 3]
```diff
@@ -8,8 +8,8 @@ export const getTransactionDailyStats = timeFrameProcedure
     openapi: {
       method: "GET",
       path: `/${TRANSACTION_BASE_PATH}`,
-      tags: ["transactions"],
-      summary: "Get transactions time series stats",
+      tags: ["stats"],
+      summary: "get transactions time series stats",
```
Comment (PJColombo) [inline packages/api/src/routers/stats/getTransactionDailyStats.ts:12]:
```suggestion
      summary: "retrieves transactions time series stats.",
```

File: packages/api/src/routers/stats/getTransactionOverallStats.ts [hunk 4]
```diff
@@ -27,8 +27,8 @@ export const getTransactionOverallStats = publicProcedure
     openapi: {
       method: "GET",
       path: `/${TRANSACTION_BASE_PATH}/overall`,
-      tags: ["transactions"],
-      summary: "Get transactions overall stats",
+      tags: ["stats"],
+      summary: "get transactions overall stats",
```
Comment (PJColombo) [inline packages/api/src/routers/stats/getTransactionOverallStats.ts:31]:
```suggestion
      summary: "retrieves transactions overall stats.",
```

File: packages/api/src/routers/tx/getAll.ts [hunk 5]
```diff
@@ -1,12 +1,32 @@
...
+  getTransactionOutputSchema,
+} from "./common";
 
 export const getAll = publicProcedure
+  .meta({
+    openapi: {
+      method: "GET",
+      path: "/transactions",
+      tags: ["transactions"],
+      summary: "get transactions",
```
Comment (PJColombo) [inline packages/api/src/routers/tx/getAll.ts:20]:
```suggestion
      summary: "retrieves all blob transactions.",
```

File: packages/api/src/routers/tx/getByHash.ts [hunk 6]
```diff
@@ -1,11 +1,24 @@
...
+} from "./common";
 import { getByHashInputSchema } from "./getByHash.schema";
 
 export const getByHash = publicProcedure
+  .meta({
+    openapi: {
+      method: "GET",
+      path: "/transactions/{hash}",
+      tags: ["transactions"],
+      summary: "get tx info",
```
Comment (PJColombo) [inline packages/api/src/routers/tx/getByHash.ts:17]:
```suggestion
      summary: "retrieves transaction details for given transaction hash.",
```

File: packages/api/src/routers/indexer/handleReorgedSlot.ts [hunk 7]
```diff
@@ -12,7 +12,7 @@ export const handleReorgedSlot = jwtAuthedProcedure
       path: `${INDEXER_PATH}/reorged-slot`,
       tags: ["indexer"],
       summary:
-        "It marks all blocks with slot greater than the new head slot as reorged.",
+        "it marks all blocks with slot greater than the new head slot as reorged.",
```
Comment (PJColombo) [inline packages/api/src/routers/indexer/handleReorgedSlot.ts:15]:
```suggestion
        "marks all blocks with slot greater than the new head slot as reorged.",
```

File: packages/api/src/routers/blockchain-sync-state/updateState.ts [hunk 8]
```diff
@@ -12,8 +12,8 @@ export const updateState = jwtAuthedProcedure
     openapi: {
       method: "PUT",
       path: BASE_PATH,
-      tags: ["blockchain-sync-state"],
-      summary: "Update the blockchain sync state",
+      tags: ["indexer"],
+      summary: "update the blockchain sync state",
```
Comment (PJColombo) [inline packages/api/src/routers/blockchain-sync-state/updateState.ts:16]:
```suggestion
      summary: "updates the blockchain sync state.",
```

File: packages/api/src/routers/blockchain-sync-state/getState.ts [hunk 9]
```diff
@@ -7,8 +7,8 @@ export const getState = publicProcedure
     openapi: {
       method: "GET",
       path: BASE_PATH,
-      tags: ["blockchain-sync-state"],
-      summary: "Get the current blockchain sync state",
+      tags: ["indexer"],
+      summary: "get the current blockchain sync state",
```
Comment (PJColombo) [inline packages/api/src/routers/blockchain-sync-state/getState.ts:11]:
```suggestion
      summary: "retrieves the blockchain sync state.",
```

File: packages/api/src/routers/blob/getAll.ts [hunk 10]
```diff
@@ -3,9 +3,19 @@ import {
...
 import { publicProcedure } from "../../procedures";
+import { getAllOutputSchema } from "./getAll.schema";
 
 export const getAll = publicProcedure
+  .meta({
+    openapi: {
+      method: "GET",
+      path: "/blobs",
+      tags: ["blobs"],
+      summary: "get blobs",
```
Comment (PJColombo) [inline packages/api/src/routers/blob/getAll.ts:14]:
```suggestion
      summary: "retrieves all blobs.",
```

File: packages/api/src/routers/blob/getByVersionedHash.ts [hunk 11]
```diff
@@ -3,12 +3,24 @@ import { TRPCError } from "@trpc/server";
...
+  getByVersionedHashOutputSchema,
+} from "./getByVersionedHash.schema";
 
 export const getByVersionedHash = publicProcedure
+  .meta({
+    openapi: {
+      method: "GET",
+      path: "/blobs/{versioned_hash}",
+      tags: ["blobs"],
+      summary: "get blob info",
```
Comment (PJColombo) [inline packages/api/src/routers/blob/getByVersionedHash.ts:17]:
```suggestion
      summary: "retrieves blob details for given versioned hash.",
```

File: packages/api/src/routers/blob/getByVersionedHash.ts [hunk 12]
```diff
@@ -3,12 +3,24 @@ import { TRPCError } from "@trpc/server";
 import type { BlobReference } from "@blobscan/blob-storage-manager";
 
 import { publicProcedure } from "../../procedures";
-import { getByVersionedHashInputSchema } from "./getByVersionedHash.schema";
+import {
+  getByVersionedHashInputSchema,
+  getByVersionedHashOutputSchema,
+} from "./getByVersionedHash.schema";
 
 export const getByVersionedHash = publicProcedure
```
Comment (PJColombo) [inline packages/api/src/routers/blob/getByVersionedHash.ts:11]: Maybe we can create an unified procedure called `getByBlobId` similar to the block id I mentioned below.

The blob id could be the versioned hash or the commitment.

Wdyt?

File: packages/api/src/routers/block/getAll.ts [hunk 13]
```diff
@@ -1,10 +1,19 @@
...
-import { getAllInputSchema } from "./getAll.schema";
+import { getAllInputSchema, getAllOutputSchema } from "./getAll.schema";
 
 export const getAll = publicProcedure
+  .meta({
+    openapi: {
+      method: "GET",
+      path: "/blocks",
+      tags: ["blocks"],
+      summary: "get blocks",
```
Comment (PJColombo) [inline packages/api/src/routers/block/getAll.ts:12]:
```suggestion
      summary: "retrieves all blocks.",
```

File: packages/api/src/routers/indexer/indexData.ts [hunk 14]
```diff
@@ -24,7 +24,7 @@ export const indexData = jwtAuthedProcedure
       method: "PUT",
       path: `${INDEXER_PATH}/block-txs-blobs`,
       tags: ["indexer"],
-      summary: "Index data in the database",
+      summary: "index data in the database",
```
Comment (PJColombo) [inline packages/api/src/routers/indexer/indexData.ts:27]:
```suggestion
      summary: "indexes data in the database.",
```

File: packages/api/src/app-router.ts [hunk 15]
```diff
@@ -23,7 +23,7 @@ export const appRouter = t.router({
       openapi: {
         method: "GET",
         path: "/healthcheck",
-        summary: "Connection healthcheck",
+        summary: "connection healthcheck",
```
Comment (PJColombo) [inline packages/api/src/app-router.ts:26]:
```suggestion
        summary: "connection healthcheck.",
```

**Review by PJColombo — CHANGES_REQUESTED** [review]
Good work!.

Left some reviews. Most of them are cosmetic typo and rephrasing changes to the endpoints' descriptions.

File: apps/rest-api-server/src/env.ts [hunk 16]
```diff
@@ -9,6 +9,10 @@ import {
 export const env = createEnv({
   envOptions: {
     server: {
+      BLOBSCAN_API_BASE_URL: z
```
[the remark this answers is not in the document]
Comment (0xGabi) [reply apps/rest-api-server/src/env.ts:12]: I was resolving this issue: https://github.com/Blobscan/blobscan/issues/205

I modified the openapi configuration here as well:https://github.com/Blobscan/blobscan/blob/public-apis/apps/rest-api-server/src/openapi.ts#L12

File: apps/web/src/pages/blob/[hash].tsx [hunk 17]
```diff
@@ -39,7 +39,7 @@ const Blob: NextPage = function () {
     isLoading,
   } = api.blob.getByVersionedHash.useQuery(
     {
-      versionedHash,
+      versioned_hash: versionedHash,
```
[the remark this answers is not in the document]
Comment (0xGabi) [reply apps/web/src/pages/blob/[hash].tsx:42]: Maybe is not relevant. Though I saw snake case was the convention on both BeaconAPI and Blockscout.

File: packages/api/src/routers/block/getByHash.ts [hunk 18]
```diff
@@ -1,11 +1,20 @@
...
+import { fullBlockSelect, getBlockOutputSchema } from "./common";
 import { getByHashInputSchema } from "./getByHash.schema";
 
 export const getByHash = publicProcedure
+  .meta({
+    openapi: {
+      method: "GET",
+      path: "/blocks/{hash}",
+      tags: ["blocks"],
+      summary: "get block info",
```
[the remark this answers is not in the document]
Comment (0xGabi) [reply packages/api/src/routers/block/getByHash.ts:13]: Yes, I agree.

File: apps/web/src/pages/blob/[hash].tsx (hunk 17 above)
[the remark this answers is not in the document]
Comment (PJColombo) [reply apps/web/src/pages/blob/[hash].tsx]: Oh, I see. Let's rename it to `hash` then.

File: apps/rest-api-server/src/env.ts (hunk 16 above)
[the remark this answers is not in the document]
Comment (PJColombo) [reply apps/rest-api-server/src/env.ts:12]: Got it. I forgot that we're gonna have an exposed api for each chain.

We can resolve this conversation then.

**Comment by Mandog1120** [comment]
a34ed3ae13dd84bccbaea7573bf0120530da830d

File: apps/rest-api-server/src/openapi.ts [hunk 19]
```diff
@@ -2,12 +2,14 @@ import { generateOpenApiDocument } from "trpc-openapi";
...
 
+import { env } from "./env";
+
 // Generate OpenAPI schema document
 export const openApiDocument = generateOpenApiDocument(appRouter, {
   title: "Blobscan API",
   description: "OpenAPI compliant REST API built using tRPC with Express",
   version: "1.0.0",
-  baseUrl: `https://api.blobscan.com/api`,
+  baseUrl: `${env.BLOBSCAN_API_BASE_URL}/api`,
```
Comment (PabloCastellano) [inline apps/rest-api-server/src/openapi.ts:12]: :+1:

**Review by PabloCastellano — APPROVED** [review]

**Pull request merged**